api/v3/utils.php

```diff
@@ -1883,35 +1926,57 @@ function _civicrm_api3_validate_html(&$params, &$fieldName, $fieldInfo) {
  * @throws Exception
  */
 function _civicrm_api3_validate_string(&$params, &$fieldName, &$fieldInfo, $entity) {
+  list($fieldValue, $op) = _civicrm_api3_field_value_check($params, $fieldName);
+  if (strpos($op, 'NULL') !== FALSE || strpos($op, 'EMPTY') !== FALSE) {
+    return;
+  }
   // If fieldname exists in params
-  $value = CRM_Utils_Array::value($fieldName, $params, '');
-  if (!is_array($value)) {
-    $value = (string) $value;
+  $fieldValue = !empty($fieldValue) ? $fieldValue : '';
+  if (!is_array($fieldValue)) {
+    $fieldValue = (string) $fieldValue;
   }
   else{
     //@todo what do we do about passed in arrays. For many of these fields
     // the missing piece of functionality is separating them to a separated string
     // & many save incorrectly. But can we change them wholesale?
   }
-  if ($value) {
-    if (!CRM_Utils_Rule::xssString($value)) {
-      throw new Exception(ts('Illegal characters in input (potential scripting attack)'));
+  
+  if ($fieldValue) {
+    if (!CRM_Utils_Rule::xssString($fieldValue)) {
+      throw new Exception('Illegal characters in input (potential scripting attack)');
     }
     if ($fieldName == 'currency') {
-      if (!CRM_Utils_Rule::currencyCode($value)) {
-        throw new Exception("Currency not a valid code: $value");
+      //When using IN operator $fieldValue is a array of currency codes
+      if (is_array($fieldValue)) {
```

Comment:
just a note - a trick Coleman taught me for dealing with not knowing if you have a string or an array is 

foreach ((array) $fieldValue as $currency) {

}

it can reduce code lines - which usually improves readability